logstash-core/src/main/java/co/elastic/logstash/api/Password.java

```diff
@@ -0,0 +1,22 @@
+package co.elastic.logstash.api;
+
+/**
+ * Wraps a password string so that it is not inadvertently printed or logged.
+ */
+public class Password {
+
+    private String password;
```

Comment:
I'm in favour of the basic API presented here.

There have been separate attempts in Logstash to ensure that we use `char[]` to hold the values of passwords, and that we clear the underlying `char[]` when the thing using the password no longer needs them. Do we need to do similar here?

Reply:
I think that would be pretty cool from a security standpoint, but I think we'd need a mechanism for the client to indicate that it was "done" with the password value and we don't have that right now.

Reply:
thankfully intermediate storage is a private implementation detail, so we can easily add that at a later time.